auth/simple.go

```diff
@@ -70,3 +117,40 @@ func (a simpleAuth) GetUser(ctx context.Context) (string, error) {
 	}
 	return id.(string), nil
 }
+
+func (a simpleAuthN) authenticate(userID string) error {
+	if userID == "" {
+		return fmt.Errorf("Must provide header: [%s]", a.userIDHeader)
+	}
+	return nil
+}
+
+func (a simpleAuthZ) authorize(httpMethod, userID string) error {
+	switch httpMethod {
+	case "GET", "":
+		return a.authorizeUser(a.whitelistReads, a.readWhitelistedUserIDs, userID)
+	case "POST", "PUT", "PATCH", "DELETE":
+		return a.authorizeUser(a.whitelistWrites, a.writeWhitelistedUserIDs, userID)
+	default:
```

Comment:
Should probably return a 404 here?

Reply:
Do you suggest http.error here or just an err message (I did the latter for now)